src/main/java/org/terasology/minion/move/MoveToAction.java

```diff
@@ -82,7 +84,7 @@ private boolean processDirect(Actor actor, MinionMoveComponent moveComponent) {
         float yaw = (float) Math.atan2(targetDirection.x, targetDirection.z);
         float requestedYaw = 180f + yaw * TeraMath.RAD_TO_DEG;
 
-        if((targetDirection.x < distance) && (targetDirection.y < distance) && (targetDirection.z < distance)) {
+        if((abs(targetDirection.x) < distance) && (abs(targetDirection.y) < distance) && (abs(targetDirection.z) < distance)) {
```

Comment:
JOML offers `distance` and `distanceSquared` methods as well as `lenght`/`lengthSquared` (distance to origin). Is it important to check each individual component or would something like this work, too?

```java
if (targetDirection.lenghtSquared() < distance * distance) {...}
```

Reply:
Something like that would definitely work. However, I think the initial idea was to save on computation time since it is called a lot of times. (Couple of ticks or so I believe). That is why I tried Implementing it that way

Reply:
We could do a detailed performance analysis here 🧐 Consider worst-case/best-case/average-case scenarios, etc. 😅 My poor complexity estimation - they should be somewhat the same (in worst case), probably the current solution is _slightly_ better on average?

- 3x `abs(...)` + 3x comparison `<`
- 3x multiplication + 2x addition + 1x comparison (`distance * distance` can be pre-computed)

Reply:
I personally like the better readability/less code of the `lengthSquared` solution...

Reply:
I have to second skal here, but not because of readability and less code, but because you're trying to calculate and check the _distance_ between to points A and B with B being the pathfinding target point and A bring the current position relative to B.
Mathematically this _distance_ is defined as the square root of the sum of the squared coordinates. So if we indeed want to check for being less than or equal to 0.2 away from the target point, we need to do the check skal proposed.
Using the current calculation/check will result in points being succeeding the check albeit being father off.

![20200717_141121.jpg](https://user-images.githubusercontent.com/29981695/87784916-7a5ea180-c837-11ea-8000-8003fbc90c01.jpg)

Reply:
Yep that totally makes sense! Added the required change